src/app/app.component.ts

```diff
@@ -80,6 +82,24 @@ export class AppComponent implements OnInit {
       });
   }
 
+  toggle() {
```

Comment:
You can reuse this function for unpin.. No need to do 2 functions, also make it inline assignment